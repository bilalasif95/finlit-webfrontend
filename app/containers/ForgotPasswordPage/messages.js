/*
 * ForgotPassword Page Messages
 *
 * This contains all the text for the ForgotPassword Page component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.ForgotPasswordPage';

export default defineMessages({
  ForgotPassword: {
    id: `${scope}.ForgotPassword`,
    defaultMessage: 'Forgot Password',
  },
  ForgotPasswordLine: {
    id: `${scope}.ForgotPasswordLine`,
    defaultMessage: `Enter your email address & we'll send you a link to reset your password`,
  },
  EmailAddress: {
    id: `${scope}.EmailAddress`,
    defaultMessage: 'Email Address',
  },
  EmailError: {
    id: `${scope}.EmailError`,
    defaultMessage: `
    Unable to reset password due to unkonown error. Please try again`,
  },
  ResetPassword: {
    id: `${scope}.ResetPassword`,
    defaultMessage: `Send Link`,
  },
  BackLogin: {
    id: `${scope}.BackLogin`,
    defaultMessage: `Back to Login`,
  },
});
