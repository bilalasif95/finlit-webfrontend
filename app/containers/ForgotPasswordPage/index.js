/*
 * Forgot Password Page
 */
import React from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { Button, FormGroup, Label, Input } from 'reactstrap';
import messages from './messages';

export default function ForgotPasswordPage() {
  return (
    <div className="registration_page">
      <Helmet>
        <title>- Forgot Password</title>
        <meta name="description" content="FinLit - Forgot Password Page" />
      </Helmet>
      <div className="form_container">
        <div className="form_content">
          <h2>
            <FormattedMessage {...messages.ForgotPassword} />
          </h2>
          <div className="form">
            <FormGroup>
              <Label for="email">
                <FormattedMessage {...messages.EmailAddress} />
              </Label>
              <Input
                type="email"
                name="email"
                id="email"
                placeholder="Your email"
              />
            </FormGroup>
            <Button>
              <FormattedMessage {...messages.SendRecoveryCode} />
            </Button>
          </div>
        </div>
      </div>
      <div className="img_container" />
    </div>
  );
}
