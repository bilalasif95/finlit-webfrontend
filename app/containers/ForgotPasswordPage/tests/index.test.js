import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router-dom';
import ForgotPasswordPage from '../index';

describe('ForgotPasswordPage />', () => {
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <BrowserRouter>
          <ForgotPasswordPage />
        </BrowserRouter>
      </IntlProvider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
