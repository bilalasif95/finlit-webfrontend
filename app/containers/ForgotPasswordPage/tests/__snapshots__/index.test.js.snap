// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordPage /> should render its heading 1`] = `
<div
  class="registration_page"
>
  <div
    class="form_container"
  >
    <div
      class="form_content"
    >
      <h2>
        <span>
          Forgot Password
        </span>
      </h2>
      <div
        class="form"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="email"
          >
            <span>
              Email Address
            </span>
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            placeholder="Your email"
            type="email"
          />
        </div>
        <button
          class="btn btn-secondary"
        >
          <span>
            Send Recovery Code
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="img_container"
  />
</div>
`;
